sdk/python/foundation-models/system/inference/fill-mask/fill-mask-batch-endpoint.ipynb

```diff
@@ -0,0 +1,470 @@
+    ")\n",
+    "workspace_ml_client.begin_create_or_update(endpoint).result()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create the deployment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "deployment_name = \"demo\"\n",
+    "\n",
+    "deployment = BatchDeployment(\n",
+    "    name=deployment_name,\n",
+    "    endpoint_name=endpoint_name,\n",
+    "    model=foundation_model.id,\n",
+    "    compute=compute_name,\n",
+    "    error_threshold=0,\n",
+    "    instance_count=1,\n",
+    "    logging_level=\"info\",\n",
+    "    max_concurrency_per_instance=2,\n",
+    "    mini_batch_size=10,\n",
+    "    output_file_name=\"predictions.csv\",\n",
+    "    retry_settings=BatchRetrySettings(max_retries=3, timeout=300),\n",
+    ")\n",
+    "workspace_ml_client.begin_create_or_update(deployment).result()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Set the deployment as default."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "endpoint = workspace_ml_client.batch_endpoints.get(endpoint_name)\n",
+    "endpoint.defaults.deployment_name = deployment_name\n",
+    "workspace_ml_client.begin_create_or_update(endpoint).wait()\n",
+    "\n",
+    "endpoint = workspace_ml_client.batch_endpoints.get(endpoint_name)\n",
+    "print(f\"The default deployment is {endpoint.defaults.deployment_name}\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 5. Run a batch inference job.\n",
+    "\n",
+    "Invoke the batch endpoint with the input parameter pointing to the folder containing the batch inference input. This creates a pipeline job using the default deployment in the endpoint. Wait for the job to complete."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input = Input(\n",
+    "    path=batch_dir,\n",
+    "    type=AssetTypes.URI_FOLDER)\n",
+    "\n",
+    "job = workspace_ml_client.batch_endpoints.invoke(\n",
+    "    endpoint_name=endpoint.name,\n",
+    "    input=input)\n",
+    "\n",
+    "workspace_ml_client.jobs.stream(job.name)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 6. Review inference predictions.\n",
+    "Download the predictions from the job output and review the predictions using a dataframe."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "scoring_job = list(workspace_ml_client.jobs.list(parent_job_name=job.name))[0]\n",
+    "\n",
+    "workspace_ml_client.jobs.download(\n",
```

Comment:
Have we tested the jobs.download() method? Is there a batchjobs.download() method?

Given that we do not populate output path in generic 'Job' contract but populate it only in 'BatchJob' contract, I wanted to make sure this method works for batch jobs.

Reply:
Not sure. I know this works, as Manoj and I have both tried it out. I see batch_endpoints has list_jobs(). Let me see if that can be leveraged.

Reply:
I am not able to get the outputs from the job name returned via list_jobs. It is the same name we already have from creating the job, so we still have to use jobs.list(parent_job_name) to get the batchscoring job, which has the "score" outputs. If you know another way, let me know.

Reply:
No problem, Amy. If we have tested that jobs.download() works, let's go with it. The only reason I asked is because recently we realized that not all job properties are populated for batch jobs unless it is retrieved as a BatchJob (rather than a generic AML Job)